service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/CreateAzureDatabaseStep.java

```diff
@@ -58,13 +59,34 @@ private String getPodName() {
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
-    azureDatabaseUtilsRunner.createDatabaseWithDbRole(
+    var cloudContext =
         context
             .getWorkingMap()
-            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class),
-        workspaceId,
-        getPodName(),
-        resource.getDatabaseName());
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+
+    // Create the database
+    azureDatabaseUtilsRunner.createDatabaseWithDbRole(
+        cloudContext, workspaceId, getPodName(), resource.getDatabaseName());
+
+    // Query LZ for the postgres server
+    var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
+    var landingZoneId =
+        landingZoneApiDispatch.getLandingZoneId(
+            bearerToken, workspaceService.getWorkspace(workspaceId));
+    var databaseResource =
+        landingZoneApiDispatch
+            .getSharedDatabase(bearerToken, landingZoneId)
+            .orElseThrow(() -> new RuntimeException("No shared database found"));
+
+    // Record resource for cleanup in Janitor
+    crlService.recordAzureCleanup(
```

Comment:
shouldn't we record cleanup before creating the database?

Reply:
Hm yeah -- will fix.